### PR TITLE
Verify that zip file entries don't try to escape the parent dir + test

### PR DESCRIPTION
**Pull request by barthanssens**
This PR addresses GitHub issue: #1210 .

Briefly describe the changes proposed in this PR:

* Check if zip file entries try to 'escape' / 'break out' of the directory  (i.e. tricks like `../../../etc/`)
* Added test to check if a file 
* Minor code formatting

### CODE REVIEW

File: util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java [hunk 1]
```diff
@@ -45,9 +50,7 @@ public static boolean isZipStream(InputStream in)
 	 * @throws IOException
 	 *         when something untoward happens during the extraction process
 	 */
-	public static void extract(File zipFile, File destDir)
-		throws IOException
-	{
+	public static void extract(File zipFile, File destDir) throws IOException {
```
Comment (hmottestad) [inline util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java]: Unrelated to the security fix. It seems you’re not following the style guide. Mind you, I think the style guide is old and terrible, designed for the screens we had in the 1980s.

File: util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java [hunk 2]
```diff
@@ -87,15 +88,16 @@ public static void extract(ZipFile zipFile, File destDir)
 	 * @throws IOException
 	 *         if the entry could not be processed
 	 */
-	public static void writeEntry(ZipFile zipFile, ZipEntry entry, File destDir)
-		throws IOException
-	{
+	public static void writeEntry(ZipFile zipFile, ZipEntry entry, File destDir) throws IOException {
 		File outFile = new File(destDir, entry.getName());
 
+		if (! outFile.getCanonicalFile().toPath().startsWith(destDir.toPath())) {
```
Comment (hmottestad) [inline util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java]: Looks like a decent way of checking for escape attempts. Btw. shouldn’t both paths be compared as canonical?

**Comment by hmottestad** [comment]
Great that you made tests!

File: util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java (hunk 1 above)
Comment (barthanssens) [reply util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java]: Thanks, I'll fix that. Probably best to switch to Eclipse IDE after all, as I'm heavily influenced by the more compact Sun Java code conventions (and Netbeans + the Eclipse-formatter don't seem to work as expected)

File: util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java (hunk 2 above)
Comment (barthanssens) [reply util/src/main/java/org/eclipse/rdf4j/common/io/ZipUtil.java]: good catch, added

**Pull request merged**